src/method/method_invoke_java.cpp

```diff
@@ -582,22 +559,13 @@ namespace cubmethod
       unpacker.unpack_all (res_code, info);
 
       query_result_info &current_result_info = info.qresult_info;
-      const auto &iter = m_cursor_map.find (current_result_info.query_id);
-      if (iter == m_cursor_map.end ())
+      query_cursor *cursor = m_group->get_cursor (current_result_info.query_id);
+      if (cursor == nullptr)
 	{
-	  query_cursor *cursor = nullptr;
-	  // not found, new cursor is created
-	  bool is_oid_included = current_result_info.include_oid;
-	  cursor = m_cursor_map[query_id] = new (std::nothrow) query_cursor (m_group->get_thread_entry(), query_id,
-	      is_oid_included);
-	  if (cursor == nullptr)
-	    {
-	      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, sizeof (query_cursor));
-	      return ER_OUT_OF_VIRTUAL_MEMORY;
-	    }
-
-	  cursor->open ();
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, sizeof (query_cursor));
+	  return ER_OUT_OF_VIRTUAL_MEMORY;
```

Comment:
Inside the m_group->get_cursor() function, there doesn't seem to be an operation to allocate memory.
How about replacing it with another appropriate error code?

Reply:
I've just changed it to return ER_FAILED. get_cursor() doesn't allocate memory.